atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/external/SchemalessStreamStoreDeleter.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.external;
+
+import java.util.Map;
+import java.util.Set;
+
+import org.immutables.value.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.Namespace;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.logging.LoggingArgs;
+import com.palantir.atlasdb.protos.generated.StreamPersistence;
+import com.palantir.atlasdb.protos.generated.StreamPersistence.StreamMetadata;
+import com.palantir.atlasdb.schema.cleanup.StreamStoreCleanupMetadata;
+import com.palantir.atlasdb.schema.stream.StreamTableType;
+import com.palantir.atlasdb.stream.GenericStreamStore;
+import com.palantir.atlasdb.transaction.api.Transaction;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.UnsafeArg;
+
+public class SchemalessStreamStoreDeleter {
+    private static final Logger log = LoggerFactory.getLogger(SchemalessStreamStoreDeleter.class);
+
+    private final Namespace namespace;
+    private final String streamStoreShortName;
+    private final GenericStreamStoreCellCreator cellCreator;
+    private final StreamStoreMetadataReader metadataReader;
+
+    public SchemalessStreamStoreDeleter(
+            Namespace namespace,
+            String streamStoreShortName,
+            StreamStoreCleanupMetadata cleanupMetadata) {
+        this.namespace = namespace;
+        this.streamStoreShortName = streamStoreShortName;
+        this.cellCreator = new GenericStreamStoreCellCreator(cleanupMetadata);
+        this.metadataReader = new StreamStoreMetadataReader(getTableReference(StreamTableType.METADATA), cellCreator);
+    }
+
+    // TODO (jkong): Main method needs tests, both unit tests and an integration test!
+    public void deleteStreams(Transaction tx, Set<GenericStreamIdentifier> streamIds) {
+        if (streamIds.isEmpty()) {
+            log.debug("deleteStreams() was called with no identifiers, so we are returning.");
+            return;
+        }
+
+        // Safe, because these are longs.
```

Comment:
You're somewhat at the mercy of the person who creates the `GenericStreamIdentifier` here; making that class (or at least its constructors) package-private would protect you here.

In general I've started biasing towards package-private where possible, to avoid exposing new classes/methods to our public API.

Reply:
Hmm, `GenericStreamIdentifier` could indeed be package private. In general, this pattern is already used in e.g. http-remoting. Will change this.